### DISCUSSION TITLE
show output of an os executed commend with auto scrolling

### DISCUSSION THREAD

**Discussion by ymolists** [charmbracelet/bubbletea#790] [Q&A] [answered]
Hi. All. My first time here. Thank you so much for making this library!

I am trying to build a docker like tool where multiple commands need to be executed in parallel. As such i cannot show all outputs at once. I want the user to see the stdout of a command by choosing from a menu. Is there an example that shows how to autoscroll a view based on input coming from a os executed command ? .i.e i dont want the user to scroll to the end.

**Comment by meowgorithm** [comment, marked as answer]
Hi! This isn't an example _per se_, but the relevant things to check out are [`ExecProcess`](https://pkg.go.dev/github.com/charmbracelet/bubbletea#ExecProcess) in Bubble Tea and [`GoToBottom`](https://pkg.go.dev/github.com/charmbracelet/bubbles@v0.16.1/viewport#Model.ScrollPercent) in Viewport.

The basic logic is:
* Use `ExecProcess` to run shell commands and get their output.
* `GoToBottom` after you get shell output.

**Reply by PylotLight** [reply]
Quick follow up question on this one.
In a similar vein, if you had a list/menu to select an option and produce output, currently using exec it just runs the command in background and goes back to menu. Is a viewport the best option for capturing/displaying that output to avoid going straight back to the menu?

Altscreen it seems may also possibly achieve what I want, how do I display menu in one screen and output of an exec command in the altscreen?
